lib/cisco_node_utils/bgp_af.rb

```diff
@@ -273,6 +273,66 @@ def dampening
       val
     end
 
+    # Return true if dampening is enabled, else false.
+    def dampening_state
+      !dampening.nil?
+    end
+
+    # For all of the following dampening getters, haf_time, reuse_time,
+    # suppress_time, and max_suppress_time, return nil if dampening
+    # is not configured, but also return nil if a dampening routemap
+    # is configured because they are mutually exclusive.
+    def dampening_half_time
+      return nil if dampening.nil? || dampening_routemap_configured?
```

Comment:
This one is a little tricky.  I have to two cases where the individual properties, half_time, reuse_time, suppress_time and max_suppress_time will be nil.

1) Dampening is not configured at all (no dampening)
2) Dampening routemap is set.

The getter returns nil when dampening is not configured and the name of the routemap when dampening routemap is configured.